tests/devices/experimental/test_default_qubit_2.py

```diff
@@ -234,38 +227,72 @@ def test_supports_backprop(self):
         """Test that DefaultQubit2 says that it supports backpropagation."""
         dev = DefaultQubit2()
         assert dev.supports_derivatives() is True
+        assert dev.supports_jvp() is True
+        assert dev.supports_vjp() is True
 
         config = ExecutionConfig(gradient_method="backprop")
         assert dev.supports_derivatives(config) is True
+        assert dev.supports_jvp(config) is True
+        assert dev.supports_vjp(config) is True
```

Comment:
I think there is some ambiguity for what makes sense with regards to backpropagation and jacobian products.  backprop is kinda a device derivative, as we aren't relying on a gradient transform and it depends on the implementation of the device. But its also not provided via a `compute_decomposition` method.  

I think it could go both ways with `supports_jvp` and `supports_vjp`.  We can always readjust the decision when we have more information later.